### DISCUSSION TITLE
recordDataTTL maybe not work

### DISCUSSION THREAD

**Discussion by snac21** [apache/skywalking#9288] [Q&A]
apm v9.1.0
I noticed that the recordDataTTL value which is defined in org.apache.skywalking.oap.server.core.CoreModuleConfig  (recordDataTTL= 7days )and application.yaml (recordDataTTL= 3days ) just opposite，and org.apache.skywalking.oap.server.storage.plugin.elasticsearch.base.HistoryDeleteEsDAO which imp IHistoryDeleteDAO used recordDataTTL as 7 days , defined in CoreModuleConfig,which can't overwite by application.yaml ,maybe i am wrong ,so ...

**Comment by wu-sheng** [comment]
I am not sure what is your point. But this field is used in here

https://github.com/apache/skywalking/blob/9793f323d582dfb87bd57949911406469b439a9d/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/ttl/DataTTLKeeperTimer.java#L112

TTL is not a thing controlled inside ES storage, it is a global setup.

**Reply by snac21** [reply]
DataTTLKeeperTimer.java,moduleConfig.getRecordDataTTL()=7day,and moduleConfig.recordDataTTL can't be overwrite,because it has not setter method ,and not relate to application.yaml core. recordDataTTL

**Reply by wu-sheng** [reply]
I don't think override configuration has anything related to `set` method. It is done through reflection.

**Reply by snac21** [reply]
maybe you are right，
=>org.apache.skywalking.oap.server.core.CoreModuleProvider#CoreModuleProvider{
this.moduleConfig = new CoreModuleConfig();
}
=>org.apache.skywalking.oap.server.core.storage.ttl.DataTTLKeeperTimer#start(getManager(), moduleConfig)

I didnot find where is done through reflection.

**Reply by snac21** [reply]
I doubt that the value recordDataTTL in the CoreModuleConfig can‘t replace by  application.yaml's recordDataTTL

**Reply by wu-sheng** [reply]
Don't doubt it. Try `debug. All codes are there. Sources are open for you already. There is no point we discussed a thing could be easily verified locally.
